### PR TITLE
Frontend

### PR DESCRIPTION
**Pull request by kenan-rhoton**
The game is now "playable" (and very very ugly).

By doing `docker-compose up --build` you can play it locally buy going to `localhost:8000` in your browser.

There is still a little problem still with Travis, but while I solve it you can do your code review.

Also, I'll put a LOT of issues on tonight (7 written down right now), so hang on tight and let's get to work!

### CODE REVIEW

File: frontend/src/components/JoinGame.vue [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+</template>
+
+<script>
+import axios from "axios";
+export default {
+  name: "JoinGame",
+  created: async function() {
+    try {
+      const gameID = this.$route.params.joinID;
+      const response = await axios.post("http://backend:3000/games/" + gameID);
```
Comment (Masclins) [inline frontend/src/components/JoinGame.vue]: I assume somehow this calls core.api/add-player
Would it be possible to comment that so it's more understandable for us plebeians?

File: backend/src/card_game/api_handler.clj [hunk 2]
```diff
@@ -5,24 +5,45 @@
...
+
+(defn parse-int [number-string]
+  (try (Integer/parseInt number-string)
+       (catch Exception e nil)))
 
 (defn play-action
-  [game player action]
-  (api/play-card-as-player game player (:index action) (:row action))
-  )
+  [game-id player-id action]
```
Comment (Masclins) [inline backend/src/card_game/api_handler.clj]: Thanks for using `game-id` and `player-id`

File: e2e/game_page.js [hunk 3]
```diff
@@ -0,0 +1,15 @@
+"use strict";
+import { Selector } from 'testcafe'; // first import testcafe selectors
+
+export default class GamePage {
+    constructor () {
+        this.hand = Selector('.hand');
+        this.cardsInHand = this.hand.find('.card');
+        this.joinLink = Selector('#join-link');
+        this.rows = Selector('.game-row');
+        this.myScore = Selector('#my-score');
```
Comment (Masclins) [inline e2e/game_page.js]: `myScore` and `opponentScore` is how many rows are won?
Given that we have 6 scores (points on each row and number of rows won) we should try to be more clear regarding that.

File: e2e/main_page.js [hunk 4]
```diff
@@ -0,0 +1,8 @@
+"use strict";
+import { Selector } from 'testcafe'; // first import testcafe selectors
+
+export default class MainPage {
+    constructor () {
+        this.createGame = Selector('#create-game');
```
Comment (Masclins) [inline e2e/main_page.js]: Where can I find `Selector('#create-game')`?

File: e2e/test.js [hunk 5]
```diff
@@ -0,0 +1,90 @@
+"use strict";
+import { Selector, ClientFunction } from 'testcafe';
+import MainPage from './main_page.js';
+import GamePage from './game_page.js';
+
```
Comment (Masclins) [inline e2e/test.js]: Not reviewing (yet) since I don't understand JavaScript tests :(

File: frontend/src/components/CreateGame.vue [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+    </div>
+</template>
+
+<script>
+import axios from "axios";
+export default {
+  name: "CreateGame",
+  beforeRouteEnter: async function(to, from, next) {
+    try {
+      const response = await axios.post("http://backend:3000/games");
```
Comment (Masclins) [inline frontend/src/components/CreateGame.vue]: I assume somehow this calls core.api/add-player
Would it be possible to comment that so it's more understandable for us plebeians?

File: e2e/game_page.js (hunk 3 above)
Comment (kenan-rhoton) [reply e2e/game_page.js]: Currently they don't work (there's no test that actually tests them) but that would be the idea.

File: e2e/main_page.js (hunk 4 above)
Comment (kenan-rhoton) [reply e2e/main_page.js]: The actual element is in frontend/src/views/Main.vue. The `#` means it's an ID selector (check out [CSS Selectors](https://www.w3schools.com/CSSref/css_selectors.asp) for more info

File: e2e/test.js (hunk 5 above)
Comment (kenan-rhoton) [reply e2e/test.js]: These aren't per se the Javascript tests (those are in frontend/test/unit/), this are the End-to-end tests, which attempt to behave like a normal user and check that things "work".

File: frontend/src/components/CreateGame.vue (hunk 6 above)
Comment (kenan-rhoton) [reply frontend/src/components/CreateGame.vue]: The interface is defined in card-game.api-handler (inside the `(defroutes app-routes` object)

**Pull request merged**